pyomo/core/tests/unit/test_units.py

```diff
@@ -0,0 +1,446 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright 2017 National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and 
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain 
+#  rights in this software.
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+#
+#
+
+import pyutilib.th as unittest
+from pyomo.environ import *
+from pyomo.core.base.template_expr import IndexTemplate
+from pyomo.core.expr import inequality
+import pyomo.core.expr.current as expr
+from pyomo.core.base.units_container import InconsistentUnitsError, UnitsError
+from six import StringIO
+
+try:
+    import pint
+    pint_available = True
+except ImportError:
+    pint_available = False
+
+def python_callback_function(arg1, arg2):
+    return 42.0
+
+@unittest.skipIf(pint_available is False, 'Testing units requires pint')
+class TestPyomoUnit(unittest.TestCase):
+
+    def test_PyomoUnit_NumericValueMethods(self):
+        m = ConcreteModel()
+        uc = units
```

Comment:
Is `units` coming in through `pyomo.environ`?

Reply:
I should have gone back and edited this comment... it is, of course.  I am wondering about the capitalization.  I think I like the lower case, but it does highlight an inconsistency in Pyomo, as the other classes / singleton symbols coming in from `pyomo.environ` are upper CamelCase.  The lower case things are functions (like `value()`) and the weird constants `maximize` and `minimize`,

Reply:
Before everything is finalized we should have a broader discussion about the singleton approach in general. There might be a better way.